Improve split with lines

bstroebl opened the pull request.

NathanW2 committed:
Move profiler instance to QgsApplication

nyalldawson committed:
Fix test for request size vs cache size

On behalf of Faunalia, sponsored by ENEL

nyalldawson committed:
Recognise that a cache can be filled using a FilterNone request

On behalf of Faunalia, sponsored by ENEL

nyalldawson committed:
Use cached feature iterator if cache has all needed features

Previously a cached feature iterator would only be returned
if cache was either full or used a cache index

On behalf of Faunalia, sponsored by ENEL

nyalldawson committed:
Increase maximum number of cached features

On behalf of Faunalia, sponsored by ENEL

nyalldawson committed:
Avoid warning noise during attribute table load

On behalf of Faunalia, sponsored by ENEL

nyalldawson committed:
Don't query list of visible features when showing selected features

On behalf of Faunalia, sponsored by ENEL

nyalldawson committed:
QgsCacheIndexFeatureId can also handle non-FilterFid requests
in certain circumstances

On behalf of Faunalia, sponsored by ENEL

nyalldawson committed:
[FEATURE] Add method to get list of unique strings matching
a substring from a vector data provider

Base implementation iterates through all features, but
providers can override with optimised versions. Native
implementations for postgres, spatialite and OGR included.

Commit:
[processing] [FEATURE] SplitWithLines

Rename algorithm SplitLinesWithLines to SplitWithLines
Accept polygon as input, too
Use only selected lines to split with (if processing is set to use selection only)
Issue log message if trying to split multi geometries
Update help

Commit:
[processing] update test for splitwithlines

nyalldawson commented on python/plugins/processing/algs/qgis/SplitWithLines.py:
```diff
@@ -0,0 +1,173 @@
+        spatialIndex = vector.spatialindex(splitLayer)
+        splitGeoms = {}
+
+        for aSplitFeature in vector.features(splitLayer):
```
You could speed this up a bit by passing a feature request to vector.features which uses `setSubsetOfAttributes([])` to avoid fetching the unused attributes.

nyalldawson commented on python/plugins/processing/algs/qgis/SplitWithLines.py:
```diff
@@ -0,0 +1,173 @@
+
+        outFeat = QgsFeature()
+        features = vector.features(layerA)
+        total = 100.0 / float(len(features))
```
you'll need to handle the cases that len(features)==0

bstroebl replied:
You're right, although this would only happen with an empty layer. This was basically a copy-paste operation from some other algo, so I expect to see this line in others, too. But will fix

nyalldawson replied:
Yeah, that's how I know to look out for it... I hit it in some other algorithm with the same block

bstroebl replied:
:-)

Commit:
[BUG] Prevent division by zero

nyalldawson commented on python/plugins/processing/algs/qgis/SplitWithLines.py:
```diff
@@ -0,0 +1,173 @@
+        outFeat = QgsFeature()
+        features = vector.features(layerA)
+        total = 100.0 / float(len(features))
+        allowedWkbTypes = [2, #WkbLineString
```
This looks fragile, and doesn't handle Z/M geometries - I'd suggest instead checking the QgsGeometry.type() for QgsWkbTypes.LineGeometry and PolygonGeometry instead, and then using QgsGeometry.isMultipart() to avoid the multipart types.

bstroebl replied:
Wouldn't it be either Line (type = 1) or Polygon (type = 2) anyways because INPUT_A is set to be either `TYPE_VECTOR_POLYGON` or `TYPE_VECTOR_LINE`?
I could skip that altogether and when handling the geometry itself check if it is multipart.

bstroebl replied:
basically I would prefer to disable multi-layers to be used as input in the first place, but there is no method to do so AFAIK.

Commit:
Speed up algorithm

Commit:
check if geometry is multipart without wkb types

Commit:
Renew test result

Commit:
[TEST] resurrect SplitLinesWithLines.py
to pass test

Commit:
rename test

Commit:
remove all splitWithLines tests

bstroebl commented:
tests have been removed to have this PR pass the checks. I will add them again in a later PR

Gustry commented:
Is this PR ready to be reviewed ?

bstroebl commented:
> Is this PR ready to be reviewed ?

Yes, from my point of view it is. Is there anything I have to do to have it reviewed? Nyall already did some reviewing (see discussion) on which I already reacted.

alexbruy committed:
[processing] add test for gdal_polygonize

alexbruy committed:
[processing] simple test for ogr buffering

m-kuhn committed:
Use QPointer for storing layer pointers in expression scope

m-kuhn committed:
Improve num_selected and is_selected function help

m-kuhn committed:
[feature] Add layer scoped actions

Within the attribute table, there is a new button to trigger actions
which are not based on individual features but instead on the whole layer.
Normally they will perform actions based on all features or the selection.

In addition to this, a lot of cleanup has been done.

alexbruy committed:
Revert "Fix extraction of ogr LayerName from multi-layer dataset URIs"

This reverts commit 6c5364186dd8d45ac51e5bd1a72c6a542f032cb1.

As it breaks all OGR geoprocessing algoroithms.

alexbruy committed:
[processing] add test for OGR polygon buffering

alexbruy committed:
[processing] add test for OGR buffer with dissolve

alexbruy committed:
[processing] fix deprecation warnings in GDAL test

alexbruy committed:
[processing] disable polygonize test, as it is not stable

alexbruy committed:
[processing] add test for points along lines

alexbruy committed:
[processing] fix OGR offset curve and one-side buffer

alexbruy committed:
[processing] add test for OGR offset lines

alexbruy committed:
[processing] test for OGR one-side buffer

alexbruy committed:
[processing] friendly names for some QGIS tests

alexbruy committed:
[processing] remove unnecessary schemas from test data

3nids committed:
[FEATURE] add functionnality to copy/move feature to move feature map tool

alexbruy committed:
[processing] use numeric parameter for buffer distance

alexbruy committed:
[processing] don't pass layer name when '-sql' parameter is used

alexbruy committed:
[processing] cleanup test data

nyalldawson committed:
[FEATURE][processing] New input type for expressions

This adds a new input type for expression inputs. Expression
inputs can be linked to a parent layer so that the builder
shows the correct fields and layer variables.

It's designed for two use cases:

1. to be used when an algorithm specifically requires an expression,
eg Select by Expression and Extract by Expression.

2. to be potentially used as a replacement input instead of string
or number literals in algorithms. Eg - if the simplify algorithm
tolerance parameter was replaced with an expression paremeter, then
this expression would be evaluated for every feature before
simplifying that feature. It would allow parameters to be calculated
per feature, as opposed to the current approach of calculating
a parameter once before running the algorithm. It would also
mean algorithms like "variable distance buffer" would no longer
be needed, as a single "buffer" algorithm could then be used
for either a fixed distance, field based, or expression based
distance.

m-kuhn committed:
Rename scopes to "Field", "Feature", "Layer"

m-kuhn committed:
Fix bindings and docs

m-kuhn committed:
Convert action ids to uuid

m-kuhn committed:
Rename listActions to actions and fix test

nirvn committed:
[FEATURE] Style management re-work and upgrade
- A new favorite grouping system was added, which the symbols list
widget defaults to
- The selected tag / smartgroup in the symbols list widget now
persists when switching layers (and across sessions)
- The symbols list widget will update the tag / smartgroup combo
box when users add / rename / remove categories
- Users can now directly tag, as well as add to favorites, symbols
while saving those to the style database
- To streamline style management, groups have been removed and
fully replaced by tags
- Tags have been integrated into the import/export user interface

nyalldawson committed:
Add QgsVectorLayer::uniqueStringsMatching() to retrieve
a list of unique strings matching a substring for a field

nirvn committed:
[FEATURE][expression] strpos() and regexp_match() improvements
- strpos() now relies on a simple string within a string search
- regexp_match() now returns pos of a matching regular expression

nyalldawson committed:
New method QgsVectorLayerUtils::createUniqueValue

Returns a new unique attribute value for a layer. For numeric
fields this is the max attribute value + 1, for strings we
attempt to create a unique value by append _1, _2, etc to either
a specified 'seed' value, or failing that to the end of the
value of that field from the first feature in the layer.

nyalldawson committed:
Fix clearing constraints

nyalldawson committed:
[FEATURE] Improve handling of defaults (inc provider default clauses,
literal defaults, and qgis expression defaults) and automatically
handle unique value constraints on layers

Add a new method QgsVectorLayerUtils::createFeature which returns
a new feature which includes all relevant defaults. Any fields
with unique value constraints will be guaranteed to have a value
which is unique for the field.

Currently only in use by the split feature tool.

Sponsored by Canton of Zug and the QGEP project

m-kuhn committed:
More api break documentation

rldhont committed:
[BUGFIX][QGIS Server] Revert layer order in WMS GetContext request

sbrunner committed:
Apply some @m-kuhn comments (#3736)

* Use QgsStringMap

* Better description

* Use new translated metadata methods

nyalldawson committed:
Make sure primary keys are always marked with a unique, not null
constraint for all providers

nyalldawson committed:
Port remaining map tools, paste features, offline editing
to use new constraint/default clause handling methods

nyalldawson committed:
[processing] When an algorithm has string parameters, also
accept numeric, file and table field inputs in modeler

This allows a non-string parameter to be reused as a string
parameter in contexts where it makes sense.

elpaso committed:
[bugfix][forwardport] File downloader for identify dialog hyperlinks

fixes #14703

Include C++ and Python tests

Travis won: ported all test cases to Python
and disabled C++ companion test (still useful locally and
for debugging)

For the curious: QTemporaryFile is not working as expected

Moved to Qt5 new style signals

Disabled C++ test and connected cancel to progress

Make string comparison on SSL errors more robust

Commit:
[virtual] Fix encoding issue

nirvn committed:
[processing] use algorithm description in modeler dependencies dialog

nyalldawson committed:
Fix QgsGeometryUtils::sqrDistToLine returns bad values (eg nan values)

nyalldawson committed:
Add method to find distance from a point to a poylgon's boundary

nyalldawson committed:
Merge pull request #3732 from nirvn/strpos_regexp_match_upgrade

[expression] strpos() and regexp_match() improvements

nyalldawson committed:
[FEATURE] Add method to calculate pole of inaccessibility for polygons

Implements a method in QgsGeometry and a processing algorithm to
calculate the pole of inaccessibility for a surface, which is the
most distant internal point from the boundary of the surface. This function
uses the 'polylabel' algorithm (Vladimir Agafonkin, 2016), which is an iterative
approach guaranteed to find the true pole of inaccessibility within a specified
tolerance. More precise tolerances require more iterations and will take longer
to calculate.

alexbruy committed:
Merge pull request #3741 from nirvn/processing_dependencies_improvement

[processing] use algorithm description in modeler dependencies dialog

alexbruy committed:
Merge pull request #3727 from arnaud-morvan/processing_ogr2ogr_defaults

[processing] fix Ogr2OgrToPostGisList algorithm

alexbruy committed:
[processing] cleanup OGR dissolve algorithm code
Avoid double extensions when creating temp files

nyalldawson committed:
Fix test failure, report distance from pole

nyalldawson committed:
Add some tags to algorithm

alexbruy committed:
[processing] improve temp file names generation

jef-n committed:
update FindQCA.cmake for Arch Linux

mhugent committed:
WMS server: add user setting if custom datasources are allowed in wms requests

mhugent committed:
Add override keywords

mhugent committed:
Merge pull request #3745 from mhugent/server_datasources

WMS server: add user setting if custom datasources are allowed in wms…

nyalldawson committed:
[FEATURE] pole_of_inaccessibility expression function

Exposes calculation of pole of inaccessiblity to expression
engine

nyalldawson committed:
[processing] Start of tests for modeler

ahuarte47 committed:
Fix iteration of features

nyalldawson committed:
Merge pull request #3746 from ahuarte47/Issue_SinglePartsToMultiparts

[FIXBUG] Fix iteration of features in SinglePartsToMultiparts algorithm

nyalldawson committed:
[processing] Restore spin box widget for number inputs outside modeller

This commit restores some pre 3.0 processing behaviour for number inputs.
Now, if a number input is required outside of modeller than a spin box
will be shown instead of a free text input.

Clicking the expression builder button results in an expression
which is evaluated immediately to avoid users expecting that
the expression will be evaluated per feature.

nyalldawson committed:
Expression parameters don't have to have parent layers

nyalldawson committed:
Add unit tests for processing expression parameter

nyalldawson committed:
Merge pull request #3703 from aaime/svg_params

Export parametric SVG, will fallback symbols for the systems that cannot understand them

nirvn committed:
[processing] check for parent alg dependencies in hasDependencies()

m-kuhn committed:
Fix Qt 5.7 build warnings

nirvn committed:
[processing] harmonize modeler UI behavior with QGIS
- use [close without saving][cancel][save] option when closing
a modeler with unsaved changes
- invert mouse wheel zoom in/out to match that of QGIS' main
canvas window
- holding the mouse middle click and dragging will pan the
modeler view

alexbruy committed:
[processing] compare only geometry in tests

alexbruy committed:
Merge pull request #3748 from nirvn/processing_harmony

[processing] harmonize modeler UI/UX behavior with QGIS

nyalldawson committed:
Allow showing 'not set' in QgsProjectionSelectionWidget

nyalldawson committed:
[processing] Use standard QGIS projection selection widget for CRS params

nyalldawson committed:
Some fixes to processing expression parameters

m-kuhn committed:
[spatialite] Don't skip default values

When inserting multiple features in a single prepared statement, the spatialite
provider would skip any default for individual features, even though they have
been specified in the field list, resulting in missing fields.

alexbruy committed:
[processing] fix parameter definition in offset curve algorithm

m-kuhn committed:
Fix more Qt5.7 warnings

alexbruy committed:
[processing] restore and fix extraction of OGR layer names

alexbruy committed:
[processing] fix indentation

alexbruy committed:
[processing] don't print Processing version

alexbruy committed:
[processing] mark buffer lines test as expected failure

This test failed as result and expected WKT geometries differ
only because of 0 vs. -0 in one coordinates pair

m-kuhn committed:
Fix memory leak

elpaso committed:
Merge pull request #3751 from elpaso/downloader_master

[bugfix][forwardport] File downloader for identify dialog hyperlinks

DelazJ committed:
typo fix

mach0 committed:
Merge pull request #3753 from DelazJ/patch-17

typo fix

nyalldawson committed:
Merge pull request #3742 from nirvn/processing_fix_hasdependencies

[processing] check for parent alg dependencies in hasDependencies()

alexbruy committed:
fix failing test

nyalldawson committed:
[processing] Add unit tests for retrieving param from script code

And fix number param retrieving from script code

nyalldawson committed:
[processing] Add button for projection selection dialog to
Crs parameters in algorithm settings in modeler

This makes it easier to pick static CRSes for the parameter,
and also makes it obvious to users that they can use a fixed
CRS parameter in their model (as opposed to one taken from
a layer or input)

wonder-sk committed:
Remove caching of WKB from QgsGeometry (was kept just for compatibility)

Also improves the API to export/import WKB as QByteArray

wonder-sk committed:
Remove caching of GEOS representation within QgsGeometry (legacy)

rouault committed:
Merge pull request #3635 from arnaud-morvan/processing_postgis_connect

[processing][db_manager] Fix connection to postgis

nyalldawson committed:
Prioritise provider default literals over reused values when
creating a new feature

nyalldawson committed:
Avoid double-evaluating postgres default values

nyalldawson committed:
Don't use auto generated widgets for fields with provider default
value clause

Using auto widgets may cause the default value clause to be mangled,
eg by converting it to a number

nyalldawson committed:
Allow empty field name ('not set' option) in QgsFieldModel, QgsFieldComboBox

nyalldawson committed:
[processing] Use standard field combo box for field choices

wonder-sk committed:
Fix oracle provider and remove dead code

nyalldawson committed:
Remove duplicate selectionChanged signal from QgsVectorLayer

If you don't want the extra info, just don't capture it...

nyalldawson committed:
Remove duplicate QgsDataProvider::dataChanged( int ) signal

Was not needed and not emitted anywhere

wonder-sk committed:
Fix conversion from WKB to string in postgres provider (follow up d729951)

alexbruy committed:
Merge pull request #3701 from alexbruy/processing-ogr-tests

[processsing] tests for GDAL algorithms

rldhont committed:
[BUGFIX][QGIS Server] Transform feature collections bbox to ESPG:4326

rldhont committed:
Fix typo

wonder-sk committed:
Propagate layer/group name changes in the layer tree (fixes #15844)

rldhont committed:
Fix QGIS Server build error

rldhont committed:
Fix setDestinationCRS

alexbruy committed:
[processing] fix test

wonder-sk committed:
Fix missing docs and sip coverage (followup 968e02d6)

wonder-sk committed:
Remove setLayerName(), layerName() from QgsLayerTreeLayer

(replaced by setName(), name() overridden from the parent)

wonder-sk committed:
Fix build

mach0 committed:
Update tsstat.pl

just a small intendation fix

nyalldawson commented on python/plugins/processing/algs/qgis/SplitWithLines.py:
```diff
@@ -0,0 +1,175 @@
+        self.addParameter(ParameterVector(self.INPUT_B,
+                                          self.tr('Split layer'), [dataobjects.TYPE_VECTOR_LINE]))
+
+        self.addOutput(OutputVector(self.OUTPUT, self.tr('Splitted')))
```
"Splitted" -> "split"

bstroebl replied:
fixed

nyalldawson commented on python/plugins/processing/algs/qgis/SplitLinesWithLines.py:
```diff
@@ -2,7 +2,7 @@
 """
 ***************************************************************************
-    SplitLines.py
+    DEPRECATED, replaced by SplitWithLines.py
```
Processing has a proper way to deprecate algorithms - see https://github.com/qgis/QGIS/blob/master/python/plugins/processing/algs/qgis/PolygonCentroids.py#L51

This will allow the algorithm to still function in existing models, but hides it away in the interface

bstroebl replied:
fixed

nyalldawson commented on python/plugins/processing/algs/qgis/SplitWithLines.py:
```diff
@@ -0,0 +1,175 @@
+        writer = self.getOutputFromName(self.OUTPUT).getVectorWriter(fieldList,
+                                                                     layerA.wkbType(), layerA.crs())
+
+        spatialIndex = vector.spatialindex(splitLayer)
```
Unfortunately this + line 71 means that all features are fetched from the provider twice - this could be a performance killer on large layers. I'd suggest you could create the spatialIndex here, and then add features to the index from within the loop at line 72. The index creation will be slower then the current approach, but the boost from only fetching features once should more than offset this.

nyalldawson replied:
Thanks! The inbuilt processing methods are fine for simple use cases (eg looping once through features and applying an operation to each), but to really optimise complex algs you need to use the  qgis api itself.

nyalldawson commented on python/plugins/processing/algs/qgis/SplitWithLines.py:
```diff
@@ -0,0 +1,175 @@
+                            if inFeatA.id() == i:
+                                continue
+
+                        if inGeom.intersects(splitGeom):
```
If you want to make this algorithm magnitudes faster you could use prepared geometries here - have a look at https://github.com/qgis/QGIS/blob/master/python/plugins/processing/algs/qgis/Union.py#L110 and https://github.com/qgis/QGIS/blob/master/python/plugins/processing/algs/qgis/Union.py#L119 for how this is done

bstroebl replied:
@Nyall i tried to implement this but get a runtime error
`TypeError: QgsGeometryEngine.intersects(QgsAbstractGeometry, str errorMsg=None): argument 1 has unexpected type 'QgsGeometry'`
api doc tells me that the first argument should be a QgsAbstractGeometryV2
The difference between my script and union is that union uses
`QgsVectorLayer.getFeatures()`
while I use processing's
`vector.features()`
which should be used, see https://github.com/qgis/QGIS/blob/master/python/plugins/processing/tools/vector.py#L93
Could that be the reason for the runtime error? If yes shouldn't the vector.features() method be improved?

nyalldawson replied:
You'll need to call `if engine.intersects(splitGeom.geometry()):`. Note the extract `.geometry()` there, which retrieves the actual geometry itself (a QgsAbstractGeometry) from within the QgsGeometry "container". It's not very nice API, but hopefully we can clean this up a bit for 3.0...

nyalldawson commented on python/plugins/processing/algs/qgis/SplitWithLines.py:
```diff
@@ -0,0 +1,175 @@
+
+                    if len(splittingLines) > 0:
+                        for splitGeom in splittingLines:
+                            splitterPList = vector.extractPoints(splitGeom)
```
you could move this after the intersects test in line 126 - if there's no intersection then you'd avoid the cost of extracting the unused points

nyalldawson replied:
If you want to optimise further - you could set splitterPList to None here, and then only extract the points when splitterPList is required (ie, if it's still None). Extracting points is slow and they may not be used if none of the intersects checks on line 126 pass.

nyalldawson commented on python/plugins/processing/tests/testdata/qgis_algorithm_tests.yaml:
```diff
@@ -191,43 +191,6 @@ tests:
         name: expected/basic_statistics_string.html
         type: file
 
-  # Split lines with lines considers two cases
```
I'm curious why these tests need to be removed - as far as I can see the deprecated qgis:splitlineswithlines algorithm hasn't been touched by this PR?

nyalldawson commented:
@bstroebl I've just posted some suggestions if you want to turbo-charge this algorithm. They are just suggestions, but I think they'd result in a dramatically faster result. :)

nyalldawson committed:
[processing] Polygons to line fixes

- Maintain Z/M values
- Handle curved geometries without segmentizing
- Retain null geometries

nyalldawson committed:
Followup 29d33b4

wonder-sk committed:
Fix rubber band in node tool not getting updated correctly

nirvn committed:
[processing] fix ExtentSelectionPanel's getValue()

nirvn committed:
[processing] fix grass/grass7 installation check

nirvn committed:
[symbology] draw effect(s) when rendering preview icon

nyalldawson committed:
Improve test debugging

rduivenvoorde committed:
minimal textual fix

aroche committed:
fix exception when selecting string function

just a typo causing an exception when selecting a string function in query builder dialog

alexbruy committed:
Merge pull request #3761 from nirvn/processing_parameterextent_fix

[processing] fix ExtentSelectionPanel's getValue()

alexbruy committed:
Merge pull request #3762 from nirvn/processing_fix_grass

[processing] fix grass/grass7 installation check

bstroebl commented:
@nyalldawson Thank you for reviewing again. I will have a look at your suggestions in the next couple days. And, yes, I am keen on speeding up the algorithm, so thank you for your hints in this respect, too.

jef-n committed:
fix build of qspatialite and ocispatial sql drivers

mach0 committed:
Add Translator

Add new italian translator

nyalldawson committed:
Add @map_extent variable containing geometry of current map extent

...Because I've seen a lot of ugly workarounds to recreate this
extent using wkt/buffers/etc

nyalldawson committed:
[processing] Use real map settings scope instead of custom canvasextent variables

nyalldawson committed:
Remove layer extent and statistic variables from processing contexts

These variables take a lot of time to calculate and cause lots of
lengthy hangs in processing. (Eg add some moderately large rasters
to a project, then try to run any processing algorithm and QGIS
will freeze).

The layer extent can already be used in expressions via the
layer_property function, which only evalutes the extent if required
and only for layers it is used for.

The band stats for raster layers should be moved to a
band_statistic function in core which behaves the same way.

nyalldawson committed:
Merge pull request #3764 from nyalldawson/cache_fix2

Cache fixes

nyalldawson committed:
Allow not set choice in QgsMapLayerComboBox

nyalldawson committed:
Allow showing CRS in QgsMapLayerComboBox

nyalldawson committed:
Allow showing additional items in QgsMapLayerComboBox

These may represent additional layers such as layers which
are not included in the map layer registry, or paths to
layers which have not yet been loaded into QGIS.

nyalldawson committed:
Allow filtering QgsMapLayerComboBox by data provider

nyalldawson committed:
[processing] Use native QGIS map layer combobox

Switches the custom map layer combo box used in the run algorithm
dialog across to the standard QGIS map layer combo box

nyalldawson committed:
[processing] Add file picker to vector/raster/table parameters
in modeller

Makes it obvious that users can set an algorithm vector/raster/
table input to a fixed table

nyalldawson committed:
Rename enum values for consistency

jef-n committed:
oracle provider: fix retrieval of column comments for geometryless tables (fixes #15853)

(cherry picked from commit a62fdb085d741da898f2bbb048c33b58aee9d630)

nyalldawson committed:
Show layer path tooltips in QgsMapLayerComboBox

Handy for differentiating between multiple layers with the
same name

alexbruy committed:
[network analysis] expose speed properter in C++ and Python API

This allows users to calculate shortest path using travel time
as optimization criteria in addition to travel distance.

jef-n committed:
fix f2e3d53 indentation

volaya committed:
Merge pull request #3747 from nyalldawson/num_input

[processing] Restore spin box widget for number inputs outside modeller

m-kuhn committed:
Disable failing PyQgsOfflineEditingWFS test

alexbruy committed:
[API][network analysis] move network analysis into analysis library to
be consistent with other analysis stuff

alexbruy committed:
remove references to networkanalysis from scrips and packaging files

alexbruy committed:
doxymentation

m-kuhn committed:
[travis] Preload libSegFault to debug crashes

nyalldawson committed:
Fix build

jef-n committed:
oracle provider: allow switching workspaces through property

(cherry picked from commit 2dc448dfb8c568c20e69fc9e3955e5df63d9a880)

nyalldawson committed:
Merge pull request #3755 from nirvn/style_manager_upgrade

[FEATURE] Style management re-work and upgrade

nyalldawson committed:
[processing] Fix error with field widget wrapper

nyalldawson committed:
[FEATURE][processing] Extract by attribute can extract for null/notnull values

Adds support for filtering where an attribute value is null or not null

nirvn committed:
[symbology] add padding value for symbol/coloramp preview

nyalldawson committed:
Merge pull request #3777 from nirvn/symbols_preview_padding

[symbology] add padding value for symbol/coloramp preview

mach0 committed:
typo fix

small typo fix

alexbruy committed:
add note to the API breaks doc

mach0 committed:
typo change Ctl to Ctrl

writing convention

m-kuhn committed:
Fix attribute table actions

m-kuhn committed:
Add a simple layer scoped action example

m-kuhn committed:
Fix some leftover faulty attribute scopes

alexbruy committed:
[API] rename QgsArcProperter to QgsStrategy

Also update subclasses names and do some more refactoring.

alexbruy committed:
update API breaks document

alexbruy committed:
add note to doxygen comments

alexbruy committed:
fix warnings

nyalldawson committed:
Fix ui build warnings

nyalldawson committed:
[FEATURE][processing] Create attribute index algorithm

Allows creation of an index on an attribute in a layer for faster
attribute based filtering

Support depends on the underlying data provider for the layer

nirvn committed:
[style manager] improve UI
- create a dedicated set of buttons for addition of tags and smartgroups
to make those actions more visible as well as getting rid of th need to
select a tag/smartgroup to create those (turns out to be quite confusing
for newcomers)
- move UI elements around, regroup {add,remove,edit} symbol buttons to
harmonize with other parts of QGIS; the elements' placement feels much
more natural now

nirvn committed:
[style manager] sort displayed symbols, insure tags are added only once

nirvn committed:
tests for two additional QgsStyle functions: tag() & symbolHasTag()

Commit:
[Irregular verb]

Commit:
[Irregular verb]

Commit:
Deprecate SplitLinesWithLines the proper way

Commit:
Deprecate SplitLinesWithLines the proper way

alexbruy committed:
Merge pull request #3783 from nyalldawson/create_attribute_index

[FEATURE][processing] Create attribute index algorithm

Commit:
[FEATURE] Speed up algo by fetching features only once

Commit:
[FEATURE] Speed up algo by fetching features only once

nyalldawson committed:
Merge pull request #3778 from nirvn/style_manager_ui_imp

[style manager] improve UI

alexbruy committed:
[processing] move custom parameter definition inside corresponding
algorithm

bstroebl commented on python/plugins/processing/algs/qgis/SplitWithLines.py:
```diff
@@ -0,0 +1,175 @@
+                for aGeom in inGeoms:
+                    passed = True
+
+                    if aGeom.wkbType == 2 or aGeom.wkbType == -2147483646:
```
I've no idea either, the Travis output might give you a clue, though (?)

bstroebl replied:
@nyalldawson You mean to only check for WKBLineString?

nyalldawson replied:
At the moment this is only passing LineString or LineString25D. I'd suggest changing it to

    if QgsWkbTypes.geometryType( aGeom.wkbType() )  == QgsWkbTypes.LineGeometry and not QgsWkbTypes.isMultiType( aGeom.wkbType() ):

Then it'll support LineStringZ, LineStringM, ...ZM, curved types, etc, and be future proof in case more are added in future.

nirvn committed:
[style dock] add minimum height to svg image list (#3786)

nirvn committed:
[style] add createMemoryDB() to QgsStyle to create temporary db

nirvn committed:
[FEATURE][style manager] import/export of symbols' tags
and favorite flag

nirvn committed:
[style manager] inform users of successful import

nyalldawson committed:
Swap order of layers in relation add dialog

Make parent layer come before child layer

nirvn committed:
use temporary memory db for style tests (avoids clash with default symbols)

nyalldawson committed:
Use standard map layer and field combo boxes in relation add dialog

alexbruy committed:
make cost() method pure virtual

alexbruy committed:
rename QgsStrategy to QgsNetworkStrategy to avoid possible future
confusion when we will have other stuff with strategies. Also
rename corresponding subclasses

alexbruy committed:
replace hardcoded magic numbers with enum

nirvn committed:
[style manager] decrease font size and add name & tags tooltip
for the import/export symbol preview list

Commit:
Use Qt5 non-OpenSSL define in testing for SSL support

Commit:
Check for Qt5 subdirectory in PYQT5_SIP_DIR

Commit:
Check for Qt5-specific version of Mac qca framework

Commit:
Update finding QCA OSSL plugin for Qt5 and C++11

Commit:
[auth] Remove dependency on SSL cert utils from auth config class

Commit:
Remove Python framework options/finding for Mac; add Homebrew support

Instead of finding Python interpreter, library and framework on Mac,
rely upon the reported paths of the interpreter (executable) to decipher
whether a framework is being used, then ensure any such framework has
its versioned subdirectory Headers used for includes and the base
Python library used directly in linking. This removes ambiguity in
framework searching, allowing just the PYTHON_EXECUTABLE (user-defined
or from FindPythonInterp module) to control which Python is used.

Commit:
Skip missing functionality in identity dialog if QWebKit not installed

Commit:
Add a CMake debug include, with simple variables dump for development

nyalldawson committed:
Merge pull request #3788 from nyalldawson/rel_dialog

Add relation dialog improvements

nyalldawson committed:
Remove redundant setLayer slots from QgsField(ComboBox|ExpressionWidget)

and switch uses of remaining slot to new style connects

Commit:
Add CMAKE_POSITION_INDEPENDENT_CODE flag to QCA test for OSSL plugin

nyalldawson committed:
Make max canvas scale 1600% (fix #15861)

Max canvas sacle should be a multiple of 2 so that zooming in
to the max and then back out again results in 100% zoom option.

Additionally, make the min/max zoom level not come from QSettings
as these aren't exposed anywhere

nyalldawson committed:
Fix ui build warning

nyalldawson committed:
Merge pull request #3789 from nirvn/style_import_export_imp

[style manager] imporve import and export experience, save symbols' tags & favorite flag

nirvn committed:
package symbology-ng-style.xml instead of the .db file

nirvn committed:
[style] improve creation of default style db
Instead of copyign a pre-existing .db shipped with QGIS, create
a new .db and import an .xml file containing our default  symbols.

On the UX front, the main benefit is being able to ship with
pre-defined tags and favorite flags for the default symbols.

On the developer side, it means we get rid of the requirement
to maintain both an .xml and a binary .db in the source tree.
We also say aurevoir to the symbol_xml2db.py script and the
need to update the (obsolete) script every time we add a new
feature to QgsStyle.

nirvn committed:
[style manager] refresh symbols shipped by default & add tags

nirvn committed:
[style] speed up importXML() by using SQL transaction
Gains are significant, importing 100 symbols would take 2.86s,
but takes only 0.18s when using transaction.

Commit:
[FEATURE] Speed up algo by using QgsGeometryEngine

Commit:
[FEATURE] Speed up algo by using QgsGeometryEngine

alexbruy committed:
rename enum and its values

alexbruy committed:
rename QgsLineVectorLayerDirector to QgsVectorLayerDirector

alexbruy committed:
remove word "road" from doxygen comments

wonder-sk committed:
Remove legacy QgsLegendInterface, move still valid methods to QgisInterface

alexbruy committed:
add doxygen note

m-kuhn committed:
Enable WFS test with a fix to prevent crash on exit (#3774)

* Revert "Disable failing PyQgsOfflineEditingWFS test"

This reverts commit 33ee514b5dfdc1b57558cb9007df5b455a59fc17.

* Let lifetime of layers be controlled by QGIS

nyalldawson committed:
Merge pull request #3773 from alexbruy/network-analysis

Network analysis

nyalldawson committed:
[processing] Fix import to postgis alg when table name not set (fix #15869, 15097)

nyalldawson committed:
[FEATURE] Expression variables for project CRS

Adds @project_crs and @project_crs_definition variables
for retrieving the current project CRS

nyalldawson committed:
[FEATURE] raster_statistic expression function for retrieving
raster band stats from a loaded layer

Allows raster band stats (eg min, max, avg) to be used in
expressions

nyalldawson committed:
Merge pull request #3766 from nyalldawson/processing_context2

[processing] Fix hangs when running algorithms

wonder-sk committed:
Remove usage of legendInterface() from plugins (followup e5f62e49)

nirvn committed:
[symbology] respect mixed unit when applying symbol from list widget (#3792)

m-kuhn committed:
Merge pull request #3793 from nirvn/default_styles_imp

[style manager] new default symbols shipped with QGIS & add a set of pre-defined tags

Commit:
[FEATURE] Broaden allowed geometries by checking WKBTypes

Commit:
[FEATURE] Broaden allowed geometries by checking WKBTypes

Commit:
[virtual] disable FilterRect when no unique id is present. Should fix #15134

Commit:
[db manager] Fix virtual layer uid handling

alexbruy committed:
fix typo

nyalldawson committed:
Fix misleading indentation warnings caused by messy qtermwidget code formatting

dakcarto committed:
Merge pull request #3790 from dakcarto/qt5py3-mac-fixes

Qt5py3 Mac fixes

nyalldawson committed:
[processing] Fix very broken delete columns algorithm

nirvn committed:
[processing] improve the modeler dialog UI and icons
- use a proper toolbar to match other parts of QGIS
- show keyboard shortcuts in toolbar action tooltips
- create vector icons to replace PNG ones

nyalldawson committed:
[processing] Enhance create points layer alg

Clean up code, allow setting z/m columns

alexbruy committed:
Merge pull request #3795 from nirvn/modeler_toolbar

[processing] improve the modeler dialog UI and icons

m-kuhn committed:
Partially revert 9bb3235

The -DCMAKE_POSITION_INDEPENDENT_CODE=ON resolves to -fPIE instead
of -fPIC (on some platforms?) rendering cmake broken.

bstroebl commented:
Now this got out of control because I tried `git rebase master`
I will create a new pull request with all changes that were included in this one but leave this open for reference for the time being
Sorry for any inconveniences

Commit:
Reactivate test for SplitWithLines

Commit:
Merge branch 'improveSplitWithLines' of github.com:bstroebl/QGIS into improveSplitWithLines
